### DISCUSSION TITLE
[Feature]: Native support for existing Traefik

### DISCUSSION THREAD

**Discussion by w4tsn** [coollabsio/coolify#3128] [Feature Requests]
### Is there an existing issue for this?

- [X] I have searched the existing issues

### Summary

Hey there :wave: I think by letting Coolify allow to specify the network of started containers and their tags while not starting / using traefik by itself native support for external traefik containers is a low-hanging fruit.

Taking this idea on a more general level: if I'm able to provide a `docker-compose` or `docker` CLI template string for the container deployment I'm free to use the full docker API with all it's features to configure my deployments. No need to re-implement the API or it's features in coolify to pass it on to docker.

### Why should this be worked on?

Tbh. I don't think handling the docker API should be a concern of Coolify, since it's purpose is automated CI/CD on a more abstract level while the specifics of the container deployment are dockers / podmans domain.

In my use-case Coolify will be another application of many which should fit into an existing infrastructure. I'd like this to work without "man-in-the-middle reconfiguration" work-arounds like proposed in a [blog post on the topic](https://beaussan.io/blog/coolify-into-existing-traefik), referenced in the docs. As stated above I think this should not be too hard to achieve, although I don't think I'll have time right now to work through the code base and implement it myself.

**Comment by yieme** [comment]
Good abstraction refinement, would be very helpful. Podman support would also be a nice bonus

**Comment by Fronix** [comment]
I've adopted a way of having an external traefik instance that handles **all** external communication and then having multiple internal instances for any other usecase.

This is then glued together using https://github.com/ldellisola/TraefikKobling which allows you to have multiple instances of traefik routed trough one. And it also allows you to have internal domains that aren't accessible externally without complex networking.

So to make an internal app external I only have to change what domain it uses. Instead of `myapp.internal.domain.com` I add `myapp.domain.com` and TraefikKobling handles the rest. Until this is solved I can recommend this approach, it's been working wonderful even together with Coolify V4.
